python/lsst/analysis/tools/actions/vector/selectors.py

```diff
@@ -325,3 +330,70 @@ def getInputSchema(self) -> KeyedDataSchema:
 
     def __call__(self, data: KeyedData, **kwargs) -> Vector:
         return cast(Vector, data[self.vectorKey.format(**kwargs)])
+
+
+# class AndSelector(VectorAction):
+#    """Apply multiple selection criteria with logical AND."""
+#
+#    selectors = ConfigurableActionStructField[VectorAction](
+#        doc="Selectors for selecting rows, will be AND together",
+#    )
+#
+#    def getInputSchema(self) -> KeyedDataSchema:
+#        for action in self.selectors:
+#            yield from action.getInputSchema()
+#
+#    def __call__(self, data: KeyedData, **kwargs) -> KeyedData:
+#        mask: Optional[Vector] = None
+#        for selector in self.selectors:
+#            subMask = selector(data, **kwargs)
+#            if mask is None:
+#                mask = subMask
+#            else:
+#                mask *= subMask  # type: ignore
+#        return cast(Vector, mask)
+
+
+class ThresholdSelector(VectorAction):
+    """Return a mask corresponding to an applied threshold."""
+
+    op = Field[str](doc="Operator name.")
+    threshold = Field[float](doc="Threshold to apply.")
+    vectorKey = Field[str](doc="Name of column")
+
+    def getInputSchema(self) -> KeyedDataSchema:
+        return ((self.vectorKey, Vector),)
+
+    def __call__(self, data: KeyedData, **kwargs) -> Vector:
+        mask = getattr(operator, self.op)(data[self.vectorKey], self.threshold)
+        return cast(Vector, mask)
+
+
+class BandSelector(VectorAction):
+    """Makes a mask for sources observed in a specified set of bands."""
+
+    vectorKey = Field[str](doc="Key of the Vector which defines the band", default="band")
+    bands = ListField[str](
+        doc="The bands to select. `None` indicates no band selection applied.",
+        default=[],
+    )
+
+    def getInputSchema(self) -> KeyedDataSchema:
+        return ((self.vectorKey, Vector),)
+
+    def __call__(self, data: KeyedData, **kwargs) -> Vector:
```

Comment:
MyPy is basically saying it cant work out what type you want bands to be, because match arms are defining it to be different things:
1) is defining it as tuple[Any, ...] - a multi element tuple of any type
2) I think it is infering the same as 1
3) it thinks it is a tuple with a single element of type Any i.e. tuple[Any]
4) None

solution to all this is before the match statement add:
`bands: tuple[Any, ...] | None`
That declares the type for the variable before it is used anywhere